examples/websearch/websearch-tools.ts

```diff
@@ -97,14 +105,19 @@ async function main() {
           const functionToCall = availableTools[toolCall.function.name]
           if (functionToCall) {
             const args = toolCall.function.arguments as any
-            console.log('\nCalling function:', toolCall.function.name, 'with arguments:', args)
+            console.log(
+              '\nCalling function:',
+              toolCall.function.name,
+              'with arguments:',
+              args,
+            )
             const output = await functionToCall(args)
             console.log('Function result:', JSON.stringify(output).slice(0, 200), '\n')
-            
+
             messages.push(chunk.message)
             messages.push({
               role: 'tool',
-              content: JSON.stringify(output),
+              content: JSON.stringify(output).slice(0, 2000 * 4), // cap at ~2000 tokens
```

Comment:
While it shortens the context length this may have quality issues as the content might be at the end. Maybe for now we can fetch one result and put it in the prompt?

Reply:
That would be valid as well but it could still be very large - i like showing that we're limited on tokens to build the understanding that the user should bump context length. but happy to do either